### PR TITLE
CLOUD-3626 - enable github actions for running bats tests

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/CLOUD-3626
Signed-off-by: Ken Wills <kwills@redhat.com>

### CODE REVIEW

File: jboss/container/wildfly/launch-config/os/added/launch/files.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ function get_maven_artifacts_jar {
   incomplete=$2
   artifacts="$(xmllint --xpath "//*[local-name()='module']/*[local-name()='resources']/*[local-name()='artifact']/@name" $moduleFile 2>/dev/null)"
 
-  read -ra ARR <<< "$artifacts"
+  readarray ARR <<< "${artifacts}"
```
Comment (luck3y) [inline jboss/container/wildfly/launch-config/os/added/launch/files.sh:22]: I suspect this is a bug, as it returns only the first element for me from $artifacts. But it was unreliable using read -ar -- perhaps its a bug in fedora's bash. readarray works every time.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: n1hility/cancel-previous-runs@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup required system packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install bats libxml2-utils nmap
+          sudo ln -s /usr/bin/ncat /usr/bin/nc
```
Comment (luck3y) [inline .github/workflows/main.yml:27]: Ubuntu 18.0.4 LTS doesn't have *ncat* which is what we really want. ncat is available in nmap, so we install that and link it to nc for the tests.

Using *netcat* from the default repos just hangs on ubuntu.

File: jboss/container/wildfly/launch/datasources/test/datasource/expectations/datasources_ibmdb2_xa.xml [hunk 3]
```diff
@@ -15,12 +15,6 @@
         <driver name="h2" module="com.h2database.h2">
             <xa-datasource-class>org.h2.jdbcx.JdbcDataSource</xa-datasource-class>
         </driver>
-        <driver name="mysql" module="com.mysql">
```
Comment (jmesnil) [inline jboss/container/wildfly/launch/datasources/test/datasource/expectations/datasources_ibmdb2_xa.xml:18]: is the changes in the config related to GitHub actions?

Comment (luck3y) [reply jboss/container/wildfly/launch/datasources/test/datasource/expectations/datasources_ibmdb2_xa.xml:18]: No, these tests needed to be updated to reflect that the config we're using to test doesn't have the mysql or postgres drivers in them anymore.

**Pull request merged**